src/components/wallet/methods/depositAsset.ts

```diff
@@ -30,25 +26,14 @@ export default async function depositAsset(
       this.account.publicKey
     )
 
-    const balances = loGet(this.account, 'state.balances')
-    const hasCurrency = loFindIndex(balances, {
-      asset_code,
-      asset_issuer,
-    })
+    this.logger.request('Fetch issuer account from Horizon', asset_issuer)
+    const issuerAccount = await this.server.loadAccount(asset_issuer)
+    this.logger.response('Fetch issuer account from Horizon', issuerAccount)
+    const homeDomain = issuerAccount.home_domain
+    if (!homeDomain) throw "Couldn't find a home_domain on the assets issuer"
 
-    if (hasCurrency === -1)
-      await this.trustAsset(asset_code, asset_issuer, pincode_stretched)
```

Comment:
Now that we use your existing balances to start the deposit flow, there's no way to get here without already having a trustline in place.

Reply:
I'm not sure I understand this. What do you mean by existing balances? Can I not provide my own account for the demo wallet to use?

Reply:
yeah you can, i just mean you can't start a deposit for a token you don't already have a trustline for, because there won't be a balance row for it which has the buttons in it.

This was originally for when the wallet had hard-coded the SRT token